File: models/UserSeason.js
import Platform from "./Platform.js";

class UserSeason {

    /**
     * @param {Object} obj
     */
    constructor(obj) {
        this.id = obj.id;
        this.addedAt = obj.added_at;
        this.platform = new Platform(obj);
    }
}

export default UserSeason;

File: models/user_season.go
package models

import (
	"anothapp-v3/entities"
	"time"
)

type UserSeason struct {
	Id       uint              `json:"id"`
	AddedAt  time.Time         `json:"addedAt"`
	Platform entities.Platform `json:"platform"`
}

func NewUserSeason(season entities.UserSeason) UserSeason {
	return UserSeason{
		Id:       season.ID,
		AddedAt:  season.AddedAt,
		Platform: season.Platform,
	}
}
